src/window/basic_frame.rs

```diff
@@ -32,15 +30,15 @@ const BUTTON_SPACE: u32 = 10;
 const ROUNDING_SIZE: u32 = 3;
 
 // defining the color scheme
-const INACTIVE_BORDER: u32 = 0xFF606060;
-const ACTIVE_BORDER: u32 = 0xFF808080;
-const RED_BUTTON_REGULAR: u32 = 0xFFB04040;
-const RED_BUTTON_HOVER: u32 = 0xFFFF4040;
-const GREEN_BUTTON_REGULAR: u32 = 0xFF40B040;
-const GREEN_BUTTON_HOVER: u32 = 0xFF40FF40;
-const YELLOW_BUTTON_REGULAR: u32 = 0xFFB0B040;
-const YELLOW_BUTTON_HOVER: u32 = 0xFFFFFF40;
-const YELLOW_BUTTON_DISABLED: u32 = 0xFF808020;
+const INACTIVE_BORDER: &[u8] = &[0x60, 0x60, 0x60, 0xFF];
+const ACTIVE_BORDER: &[u8] = &[0x80, 0x80, 0x80, 0xFF];
+const RED_BUTTON_REGULAR: &[u8] = &[0x40, 0x40, 0xB0, 0xFF];
+const RED_BUTTON_HOVER: &[u8] = &[0x40, 0x40, 0xFF, 0xFF];
+const GREEN_BUTTON_REGULAR: &[u8] = &[0x40, 0xB0, 0x40, 0xFF];
+const GREEN_BUTTON_HOVER: &[u8] = &[0x40, 0xFF, 0x40, 0xFF];
+const YELLOW_BUTTON_REGULAR: &[u8] = &[0x40, 0xB0, 0xB0, 0xFF];
+const YELLOW_BUTTON_HOVER: &[u8] = &[0x40, 0xFF, 0xFF, 0xFF];
+const YELLOW_BUTTON_DISABLED: &[u8] = &[0x20, 0x80, 0x80, 0xFF];
```

Comment:
Okay, I didn't expect `byteorder` to be such a bottleneck.

However, hard-coding the values like this is kind of wrong, given the wayland protocol specifies that values should be in native endianness. So this would be wrong if run in a big endian architecture (not sure if that could happen though ?). In all cases, to be safe, I'd rather keep handling endianness correctly. Things like `#[cfg(target_endian = "little")]` should allow this I believe.

Reply:
I chose to keep the constants in little endian and reverse if big because the device will more then likely be little endian so its faster

Reply:
I was rather thinking about defining the constants using `#[cfg(...)]` to avoid runtime costs in all cases, so something like:

```rust
#[cfg(target_endian = "little")]
mod colors {
    /* color definitions in little endian */
}
#[cfg(target_endian = "big")]
mod colors {
    /* color definitions in big endian */
}
use self::colors::*;
```